lib/internal/modules/esm/load.js

```diff
@@ -108,6 +132,51 @@ async function defaultLoad(url, context = kEmptyObject) {
     source,
   };
 }
+/**
+ * @typedef LoadContext
+ * @property {string} [format] Hint possibly returned by `resolve`
+ * @property {string | Buffer | ArrayBuffer} [source] source
+ * @property {Record<string, string>} [importAssertions] import attributes
+ */
+
+/**
+ * @typedef LoadReturn
+ * @property {string} format format
+ * @property {string} responseURL url
+ * @property {Buffer} source source
+ */
+
+/**
+ * @param {string} url
+ * @param {LoadContext} [context]
+ * @returns {LoadReturn}
+ */
+function defaultLoadSync(url, context = kEmptyObject) {
+  let responseURL = url;
+  const { importAssertions } = context;
+  let {
+    format,
+    source,
+  } = context;
+
+  format ??= defaultGetFormat(new URL(url), context);
```

Comment:
Format does not / should not change, it just needs a default value, so I think it would be better to keep it a constant with a default:

```suggestion
  const {
    format = defaultGetFormat(new URL(url), context),
    importAssertions,
  } = context;

  let { source } = context;
```

Reply:
What if `format` is `null`?

Reply:
Ah, we have a bug in the `resolve` hook final result validation: https://github.com/nodejs/node/blob/d150316a8ecad1a9c20615ae62fcaf4f8d060dcc/lib/internal/modules/esm/hooks.js#L336-L339

`format` is an enum, so `null` is not valid (only the enum values or nothing)

Let's fix them together (we can't fix this part without creating a second bug).

Reply:
I'm pretty sure we've always accepted nullish values indifferently in the hook APIs.

Reply:
Even for enums? Why? `null` is an actual value.